docs/ext/pydantic_autosummary/generate.py

```diff
@@ -0,0 +1,880 @@
+"""Generates reST source files for autosummary.
+
+Usable as a library or script to generate automatic RST source files for
+items referred to in autosummary:: directives.
+
+Each generated RST file contains a single auto*:: directive which
+extracts the docstring of the referred item.
+
+Example Makefile rule::
+
+   generate:
+           sphinx-autogen -o source/generated source/*.rst
+"""
+
+from __future__ import annotations
+
+import argparse
+import importlib
+import inspect
+import locale
+import os
+import pkgutil
+import pydoc
+import re
+import sys
+from os import path
+from typing import TYPE_CHECKING, Any, NamedTuple
+
+import sphinx.locale
+from jinja2 import TemplateNotFound
+from jinja2.sandbox import SandboxedEnvironment
+from sphinx import __display_version__
+from sphinx.builders import Builder
+from sphinx.config import Config
+from sphinx.ext.autodoc.importer import import_module
+from sphinx.locale import __
+from sphinx.pycode import ModuleAnalyzer, PycodeError
+from sphinx.registry import SphinxComponentRegistry
+from sphinx.util import logging, rst
+from sphinx.util.inspect import getall, safe_getattr
+from sphinx.util.osutil import ensuredir
+from sphinx.util.template import SphinxTemplateLoader
+
+# ----------- pydantic_autosummary change
+from . import (
+    ImportExceptionGroup,
+    get_documenter,
+    import_by_name,
+    import_ivar_by_name,
+)
+from .pydantic import set_pydantic_model_fields
+
+# ----------/ pydantic_autosummary change
```

Comment:
This file is also a direct copy from autosummary, and all changes I made are marked here.

The first import set just made the path relative over `sphinx.autosummary`, but importing `from .pydantic import set_pydantic_model_fields` was my addition.